README.md

```diff
@@ -28,29 +28,28 @@ Then, import the package in your Dart code:
 import 'package:friends_badge/friends_badge.dart';
 ```
 
-Now, you can use the `NfcBadgeRepository` to write images to your badge:
+Now, you can use the `FriendsBadge` class to create a `BadgeImage` and write it
+to your badge:
 
 ```dart
-final repository = NfcBadgeRepository();
-
-final image = img.decodeImage(yourImageBytes);
-if (image != null) {
-  repository
-      .writeOverNfc(image)
-      .listen(
-        (progress) {
-          // Update progress UI
-        },
-        onError: (error) {
-          // Handle error
-        },
-        onDone: () {
-          // Write operation completed
-        },
-      );
-}
+// First, create a BadgeImage from an image.
+final image = FriendsBadge.createBadgeImage(yourImage);
+
+// Then, write the image to the badge over NFC.
+await WaitingForNfcTap.showLoading(
+  context: context,
+  job: FriendsBadge.nfcBadgeRepository.writeOverNfc(
```

Comment:
This feels a bit unergonomic. :thinking: 
I would have a static class that creates a singleton that it can use and then you can pass in an image, so that the final API for the user is something like this `FriendsBadge.writeOverNfc(image)`.
For this type of API I don't think that the user should need to know about the repository.